alicloud/resource_alicloud_slb_acl.go

```diff
@@ -0,0 +1,316 @@
+			"frontend_port":    i.ListenerPort,
+			"acl_type":         i.AclType,
+		}
+		result = append(result, l)
+	}
+
+	return result
+}
+
+func resourceAlicloudSlbAcl() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudSlbAclCreate,
+		Read:   resourceAlicloudSlbAclRead,
+		Update: resourceAlicloudSlbAclUpdate,
+		Delete: resourceAlicloudSlbAclDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+				Default:  "tf-slb-acl",
+			},
+
+			"ip_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "ipv4",
+			},
+			"entrys": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"entry": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"comment": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+				MaxItems: 300,
+				MinItems: 0,
+			},
+			"related_listeners": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"load_balancer_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"frontend_port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"acl_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"protocol": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				MaxItems: 300,
+				MinItems: 0,
+			},
+		},
+	}
+}
+
+func resourceAlicloudSlbAclCreate(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*AliyunClient).slbconn
+
+	name := strings.Trim(d.Get("name").(string), " ")
+	ip_version := strings.Trim(d.Get("ip_version").(string), " ")
+
+	request := slb.CreateCreateAccessControlListRequest()
+	request.AclName = name
+	request.AddressIPVersion = ip_version
+
+	response, err := client.CreateAccessControlList(request)
+	if err != nil {
+		if IsExceptedError(err, SlbAclInvalidActionRegionNotSupport) {
+			return fmt.Errorf("CreateAccessControlList got an error: %#v", err)
+		}
+		if IsExceptedError(err, SlbAclNumberOverLimit) {
+			return fmt.Errorf("CreateAccessControlList got an error: %#v", err)
+		}
+		return fmt.Errorf("CreateAccessControlList got an error: %#v", err)
+	}
+
+	d.SetId(response.AclId)
+
+	if entryList, ok := d.GetOk("entrys"); ok {
+		list := entryList.(*schema.Set).List()
+		if len(list) > 0 {
+			if err := slbAddAccessControlListEntry(client, list, response.AclId); err != nil {
+				return err
+			}
+		}
+	}
+
+	return resourceAlicloudSlbAclRead(d, meta)
+}
```

Comment:
Please remove line 218-225 and line 227 should return return resourceAlicloudSlbAclUpdate(d, meta)